whole-tale/girder_wholetale
# Add wholetale plugin configuration settings for basic branding

This PR updates the wholetale plugin to add configuration settings and customizations for basic branding.  This includes:

- Adds `/wholetale/settings` endpoint to return current settings:
    - `core.brand_name` (DASHBOARD_TITLE): Dashboard and Girder title
    - `core.banner_color` (HEADER_COLOR): Dashboard and Girder.header color
   - `wholetale.website_url`: URL for logo/title link
    - `wholetale.dashboard_link_title`: Title for dashboard link displayed in Girder
    - `wholetale.catalog_link_title`: Title for (optional) data catalog link displayed on the dashboard
    - `wholetale.enable_data_catalog`: Whether to display the data catalog link
    - `wholetale.dashboard_url`: URL for the dashboard (from env)
    - `wholetale.logo`: Logo displayed in Dashboard and Girder header
- Updates Girder configView for the above
- Updates Girder header to use the above settings

To test:
- Checkout this branch and https://github.com/whole-tale/ngx-dashboard/pull/324
- Rebuild dashboard and plugins if needed
- As Girder admin:
  - Admin console > server configuration
      - Change brand name to "My Brand" and banner color to hot pink
  - Admin console > plugins > Whole Tale
      - Change dashboard website URL to https://google.com
      - Change dashboard link title  to "My Dashboard"
      - Select "enable data catalog" and change data catalog link title to "My Data"
      - Upload a new dashboard logo
  - Save
- Refresh dashboard UI and Girder UI
   - Confirm dashboard and UI have matching logo, header color, and brand nam
   - Confirm Girder UI displays a link "My Dashboard" that takes you to dashboard.* when clicked
   - Confirm Dashboard UI displays a link "My Data" that takes you to girder.* when clicked
   - Confirm dashboard logo/brand have link https://google.com  
- In whole tale plugin settings, uncheck Enable data catalog and save
   - Confirm data catalog link is no longer displayed in dashboard.
